lib/surface/components/form/hidden_input.ex

```diff
@@ -21,13 +21,13 @@ defmodule Surface.Components.Form.HiddenInput do
   import Surface.Components.Form.Utils
 
   def render(assigns) do
-    form = get_form(assigns)
-    field = get_field(assigns)
     props = get_non_nil_props(assigns, [:value, :class])
     event_opts = get_events_to_opts(assigns)
 
     ~H"""
-    {{ hidden_input(form, field, props ++ @opts ++ event_opts) }}
+    <InputContext assigns={{ assigns }} :let={{ form: form, field: field }}>
```

Comment:
why the full assigns instead of a prop for form and a prop for field?

Reply:
Well, since we don't have to worry about change tracking for form components and this is kind of a "private" component, I think making the whole thing available would be more flexible so we can add new features easily in the future.